File: pkg/worker/retry.go
package worker

import "context"

type RetryDecision struct {
	Retry bool
	Nack  bool
}

type RetryPolicy interface {
	OnError(ctx context.Context, evt *Event, err error) RetryDecision
}

type NoRetry struct{}

func (NoRetry) OnError(ctx context.Context, evt *Event, err error) RetryDecision {
	return RetryDecision{Retry: false, Nack: true}
}

File: sdk/typescript/worker/src/retry.ts
import type { WorkerContext } from "./context.js";
import type { Event } from "./event.js";

export interface RetryDecision {
  retry?: boolean;
  nack?: boolean;
  Retry?: boolean;
  Nack?: boolean;
}

export interface RetryPolicy {
  onError?: (ctx: WorkerContext, evt: Event | undefined, err: Error) => RetryDecision;
  OnError?: (ctx: WorkerContext, evt: Event | undefined, err: Error) => RetryDecision;
}

export class NoRetry implements RetryPolicy {
  onError(_ctx: WorkerContext, _evt: Event | undefined, _err: Error): RetryDecision {
    return { retry: false, nack: true };
  }
}

export function normalizeRetryDecision(decision: RetryDecision): { retry: boolean; nack: boolean } {
  return {
    retry: Boolean(decision.retry ?? decision.Retry),
    nack: Boolean(decision.nack ?? decision.Nack),
  };
}
